packages/syft/tests/syft/core/node/common/service/user_manager_service_benchmark_test.py

```diff
@@ -53,18 +54,27 @@
 from syft.core.node.domain_service import DomainServiceClass
 
 
-def _create_dummy_user(faker: Faker, is_admin=False):
+def _create_dummy_user(
+    faker: Faker, domain: DomainInterface, is_admin: bool = False
+) -> Tuple[dict, VerifyKey]:
 
     # Create dummy user
+    private_key = SigningKey.generate()
+    verify_key = private_key.verify_key
+
+    # Check if generated email is already been used for another account previously registered.
+    email = faker.free_email()
+    while domain.users.contain(email=email):
```

Comment:
Q: should it be an if statement?

Reply:
The `while ` statement is to ensure that the randomly generated email has not yet been used by any previously registered accounts.

Reply:
While I was tidying these tests I came across situations where the same email was generated at two different times causing failures in the tests.

To ensure that this does not happen, the `while` loop will only allow the exit when we have an email that has never been used in any previously registered account.

Reply:
Ahh ok, it creates a new email on each function call, I initially thought faker was global and would provide same email each time 👍